### PR TITLE
Fix example code in Doc

### PR DESCRIPTION
**Pull request by ephraimbuddy**
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Comment by uranusjr** [comment]
How about making this conform to PEP 8 style? (four spaces, no space around parameter `=`)

File: docs/apache-airflow/concepts/params.rst [hunk 1]
```diff
@@ -33,7 +33,7 @@ which won't be doing any such validations.
 
     with DAG(
       'my_dag',
-      params: {
+      params = {
```
Comment (kaxil) [inline docs/apache-airflow/concepts/params.rst]:
```suggestion
      params={
```

File: docs/apache-airflow/concepts/params.rst [hunk 2]
```diff
@@ -32,21 +32,22 @@ which won't be doing any such validations.
...
-            default='example@example.com',
-            type='string',
-            format='idn-email',
-            minLength=5,
-            maxLength=255,
-        ),
-      }
+        'my_dag',
+        params={
+          'int_param': Param(10, type='integer', minimum=0, maximum=20),          # a int param with default value
```
Comment (uranusjr) [inline docs/apache-airflow/concepts/params.rst]: Increase indent here

File: docs/apache-airflow/concepts/params.rst [hunk 3]
```diff
@@ -32,21 +32,22 @@ which won't be doing any such validations.
...
+            'old_param': 'old_way_of_passing',                                      # i.e. no data or type validations
+            'simple_param': Param('im_just_like_old_param'),                        # i.e. no data or type validations
+            'email_param': Param(
+                default='example@example.com',
+                type='string',
+                format='idn-email',
+                minLength=5,
+                maxLength=255,
+            ),
+          },
```
Comment (uranusjr) [inline docs/apache-airflow/concepts/params.rst]:
```suggestion
        params={
            'int_param': Param(10, type='integer', minimum=0, maximum=20),  # a int param with default value
            'str_param': Param(type='string', minLength=2, maxLength=4),    # a mandatory str param
            'dummy_param': Param(type=['null', 'number', 'string'])         # a param which can be None as well
            'old_param': 'old_way_of_passing',                              # i.e. no data or type validations
            'simple_param': Param('im_just_like_old_param'),                # i.e. no data or type validations
            'email_param': Param(
                default='example@example.com',
                type='string',
                format='idn-email',
                minLength=5,
                maxLength=255,
            ),
        },
```

**Review by uranusjr — APPROVED** [review]

**Pull request merged**